### PR TITLE
DR-168 Implement base exceptions and global exception handler

### PR DESCRIPTION
**Pull request by ddietterich**
- add base exceptions in top-level exception package
 - revise all other exceptions to derive from base exceptions
 - create new exceptions (derived from base exceptions) for everything we throw in the datarepo code except stairway.
 - add global exception handler; everything is turned into ErrorModel now
 - set application.properties to throw on bad URL so we can control the response to that error

### CODE REVIEW

File: src/main/java/bio/terra/pdao/PrimaryDataAccess.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
  *     </ol>
  * </ul>
  *
- * All implementation-specific exceptions should be caught by the implementation and wrapped
+ * All implementation-specific exception should be caught by the implementation and wrapped
```
Comment (ahaessly) [inline src/main/java/bio/terra/pdao/PrimaryDataAccess.java]: should still be plural - probably a search/replace mistake

File: src/main/java/bio/terra/upgrade/MigrateException.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package bio.terra.upgrade;
 
 /**
- * Migrate exception is thrown on Liquibase or SQL exceptions during migration.
+ * Migrate exception is thrown on Liquibase or SQL exception during migration.
```
Comment (ahaessly) [inline src/main/java/bio/terra/upgrade/MigrateException.java]: also revert

**Review by ahaessly — APPROVED** [review]
2 comment changes

**Review by thathert — APPROVED** [review]
Good to go once Andrea's comments are addressed

**Pull request merged**